File: codegen/features/xsls/backendless-codegen-xsls/ios/backendless-codegen/ObjC/CRUD/CRUD/ViewControllers/ObjectsWithRelationsViewController.m

#import "ObjectsWithRelationsViewController.h"
#import "ProperiesViewController.h"
#import "Utils.h"
#import <Backendless-Swift.h>

#define FIND_RELATED @"Find with related"
#define SHOW_PROPERTIES @"ShowRelatedProperties"

@implementation ObjectsWithRelationsViewController

- (void)viewDidLoad {
    [super viewDidLoad];
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return self.objectIds.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"ObjectIdCell" forIndexPath:indexPath];
    cell.textLabel.text = [self.objectIds objectAtIndex:indexPath.row];
    return cell;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    [self performSegueWithIdentifier:SHOW_PROPERTIES sender:indexPath];
}

- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    ProperiesViewController *propertiesVC = [segue destinationViewController];
    propertiesVC.operation = FIND_RELATED;
    propertiesVC.related = self.related;
    propertiesVC.objectId = [self.objectIds objectAtIndex:((NSIndexPath *)sender).row];
    propertiesVC.tableName = self.relatedTableName;
    
    [Backendless.shared.data describeWithTableName:self.relatedTableName responseHandler:^(NSArray *properties) {
        NSMutableDictionary<NSString *, NSString *> *fields = [NSMutableDictionary<NSString *, NSString *> new];
        for (ObjectProperty *property in properties) {
            if (![[property getTypeName] isEqualToString:@"RELATION"] &&
                ![[property getTypeName] isEqualToString:@"RELATION_LIST"]) {
                [fields setObject:[property getTypeName] forKey:property.name];
            }
        }
        propertiesVC.fields = fields;
        [propertiesVC.tableView reloadData];
    } errorHandler:^(Fault *fault) {
        [utils showErrorAlert:self fault:fault];
    }];
}

@end

File: codegen/features/xsls/backendless-codegen-xsls/ios/backendless-codegen/Swift/CRUD/CRUD/ViewControllers/ObjectsWithRelationsViewController.swift

import UIKit
import Backendless

class ObjectsWithRelationsViewController: UITableViewController {
    
    var tableName = ""
    var objectIds = [String]()
    var related = ""
    var relatedTableName = ""
    var propertyType = ""
    
    private let FIND_RELATED = "Find with related"
    private let SHOW_PROPERTIES = "ShowRelatedProperties"
    
    private let utils = Utils.shared
    
    override func viewDidLoad() {
        super.viewDidLoad()
    }
    
    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return objectIds.count
    }
    
    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "ObjectIdCell", for: indexPath)
        cell.textLabel?.text = objectIds[indexPath.row]
        return cell
    }
    
    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        performSegue(withIdentifier: SHOW_PROPERTIES, sender: indexPath)
    }
    
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        if let propertiesVC = segue.destination as? PropertiesViewController,
            let indexPath = sender as? IndexPath {
            propertiesVC.operation = FIND_RELATED
            propertiesVC.related = related
            propertiesVC.objectId = objectIds[indexPath.row]
            propertiesVC.tableName = relatedTableName
            Backendless.shared.data.describe(tableName: relatedTableName, responseHandler: { properties in
                var fields = [String : String]()
                for property in properties {
                    if property.getTypeName() != "RELATION" && property.getTypeName() != "RELATION_LIST" {
                        fields[property.name] = property.getTypeName()
                    }
                }
                propertiesVC.fields = fields
                propertiesVC.tableView.reloadData()
            }, errorHandler: { [weak self] fault in
                if self != nil {
                    self?.utils.showErrorAler(target: self!, fault: fault)
                }
            })
        }
    }
}

